web/api/login-callback/index.ts

```diff
@@ -149,6 +154,19 @@ export const loginCallback = withApiAuthRequired(async (req: NextRequest) => {
     );
   }
 
+  // ANCHOR: Auth0 account linking
+  if (user.world_id_nullifier && user.email && user.auth0Id) {
+    try {
+      await linkAccounts({
+        email: user.email,
```

Comment:
@andy-t-wang In theory it can be done right in the place where we are adding an additional auth method. But it will work only if the initial auth method was Sign in with worldcoin and we are adding an email method. In this case, we are [starting email verification](https://github.com/worldcoin/developer-portal/pull/666/files#diff-1c2dbfbdb511dcdb5296f07ad17bcaed3969764ea20d5693885021e28dbc055eR34) with auth0 and it will create an email auth0 account that can be linked to the primary auth0 account right after we did this. 

But in the case where email was primary auth method we can only simulate world id verification to get nullifier hash to put it in the hasura DB. We can't create worldcoin type auth0 account by API, it just throwing an error, when you try to do this AND also you can't just add worldcoin account identity directly to the primary email account. Because additional identity should be created from existing auth0 account. 

**TLDR;** So currently we are doing it on login because in case where email auth method is primary we need to create worldcoin type auth0 account first to connect it as identity to the primary account later. It can be done on login only, as I can see.

Reply:
To be honest I think account linking only makes sense if another account already exists. This is like pre-emptive linking.

Reply:
I guess in the ultimate ideal case now that I think about it we should have users link accounts that exist and merge the data. This feels hacky otherwise

Reply:
> makes sense if another account already exists
Ok, let's think that we will implement something that will merge accounts for the users that already have two accounts that were created before. 

What the flow should look like for the new user? Should new users create two accounts also and only then our app will suggest merging them?

Reply:
Yeah I think that makes more sense. No pre-emptive merging. We let users merge two existing accounts. The challenge will be with permissions though  since this will require editing teams that you're not a part of

Reply:
I think merging makes the most sense. For example Paolo has two accounts that need to be merged and are already created.